ProjFS.Mac/PrjFSKextLogDaemon/PrjFSKextLogDaemon.cpp

```diff
@@ -221,12 +239,141 @@ static bool TryFetchAndLogKextHealthData(io_connect_t connection)
             healthData.totalFindRootForVnodeMisses,
             healthData.totalRefreshRootForVnode,
             healthData.totalInvalidateVnodeRoot);
+        
+        WriteJsonToMessageListener(
+        "{"
+            "\"Message\":\"Vnode cache health\","
+            "\"CacheCapacity\":\"" + to_string(healthData.cacheCapacity) + "\","
+            "\"CacheEntries\":\"" + to_string(healthData.cacheEntries) + "\","
+            "\"InvalidationCount\":\"" + to_string(healthData.invalidateEntireCacheCount) + "\","
+            "\"CacheLookups\":\"" + to_string(healthData.totalCacheLookups) + "\","
+            "\"LookupCollisions\":\"" + to_string(healthData.totalLookupCollisions) + "\","
+            "\"FindRootHits\":\"" + to_string(healthData.totalFindRootForVnodeHits) + "\","
+            "\"FindRootMisses\":\"" + to_string(healthData.totalFindRootForVnodeMisses) + "\","
+            "\"RefreshRoot\":\"" + to_string(healthData.totalRefreshRootForVnode) + "\","
+            "\"InvalidateRoot\":\"" + to_string(healthData.totalInvalidateVnodeRoot) + "\","
+        "}");
```

Comment:
I trust all of the above values are integers, so we don’t need to worry about escape/quoting them.

Reply:
If they are integers, you can omit the dquotes as in “key”:123